testsuite/gnat2goto/tests/floating_point_constant/floating_point_constant.adb

```diff
@@ -0,0 +1,17 @@
+--  This test is testing the floating point constants
+procedure Floating_Point_Constant is
```

Comment:
We already have `floating_point_literals`

Reply:
1. In case relevant:
Literals and constants are significantly different things in C
https://stackoverflow.com/questions/11732795/are-the-literal-and-constant-the-same-concept-in-c

2. Should we be testing the 'constancy' aspect here?  (which a literal is, but a const integer is not necessarily a literal). 

Found this which seems interesting:
As a result, numeric literals in Ada do not carry with them their type as in C. The actual type of the literal is determined from the context.
https://www.adacore.com/gems/ada-gem-7
So the C representation of an ada literal will be ... interesting. But an Ada const?

I have only skim read enough to go: "oh!". Not enough time atm to do more.

Also:
"I'm reading Norman Cohen's Ada 95 book and on page 129 we have the constant declarations:

 Pi: constant Float := 3.1415926536
and

Pi: constant := 3.1415926536
The second declaration is said to be interpreted as, I quote: "any floating-point of fixed-point type with the appropriate range". "
https://stackoverflow.com/questions/3178339/ada-constant-declaration

So here we'd want the first. Not the second (which is a literal). 
My interpretation - from skimming.

Reply:
Ah right, but in this case it's just us using terminology incorrectly/inconsistently. `Floating_Point_Literals` is checking that we can deal with floating point literals _and_ assign them to variables _and_ have cbmc understand expressions involving floating point stuff

Reply:
So this 
procedure Floating_Point_Literals is
  X : Float := 1.0;

Is really just floating point number initialisation - not a 'literal' ?

"it can be said that the term literal in C stands for unnamed object that occupies memory (literals are typically lvalues)"
https://stackoverflow.com/questions/11732795/are-the-literal-and-constant-the-same-concept-in-c